aten/src/ATen/native/TensorIterator.h

```diff
@@ -478,7 +488,9 @@ class CAFFE2_API TensorIteratorConfig final {
   // It would be better if this was && qualified, but this would be at the cost
   // of a lot of boilerplate above
   TensorIterator build() {
-    return TensorIterator(*this);
+    TensorIterator iter;
```

Comment:
did you kill the `TensorIterator(TensorIteratorConfig&)` constructor just to clean things up?

Reply:
You are right, I don't think I was supposed to delete this constructor, and I ended up just refactoring around the problem. Not sure if it's worth restoring it though!

Reply:
Yeah I think this is better - more transparent. The other just hid the build call